src/sfdt/toggleBookmark.ts

```diff
@@ -5,6 +5,88 @@ import {
 	isBookmarkStart,
 	isBookmarkEnd,
 } from '../queryBookmark'
+import Stack from '../stack'
```

Comment:
i dont think implementing a custom stack adds any benefits. 

perhaps peekAt and pop are helpful to save code, but they can easily be implemented here as functional helpers, rather than a new oop construction.

eg:
```
const pop = (items) => items.length === 0 ? 'Underflow' : items.pop()
```

though i worry about this function having two different return types, that is generally not good practise.

ps. take note to use the Zegal JS style of single quotes, triple equals and always having brackets around function params.